core/bootstrap/src/main/java/org/eclipse/dataspaceconnector/system/runtime/BaseRuntime.java

```diff
@@ -0,0 +1,183 @@
+        monitor.severe("Error booting runtime", e);
+        System.exit(-1);  // stop the process
+    }
+
+    /**
+     * Starts all service extensions by invokint {@link org.eclipse.dataspaceconnector.system.ExtensionLoader#bootServiceExtensions(List, ServiceExtensionContext)}
+     *
+     * @param context           The {@code ServiceExtensionContext} that is used in this runtime.
+     * @param serviceExtensions a list of extensions
+     */
+    protected void bootExtensions(ServiceExtensionContext context, List<ServiceExtension> serviceExtensions) {
+        bootServiceExtensions(serviceExtensions, context);
+    }
+
+    /**
+     * Create a list of {@link ServiceExtension}s. By default this is done using the ServiceLoader mechanism. Override if
+     * e.g. a custom DI mechanism should be used.
+     *
+     * @param context A context to which all the service extensions should be registered
+     * @return a list of {@code ServiceExtension}s
+     */
+    @NotNull
+    protected List<ServiceExtension> createExtensions(ServiceExtensionContext context) {
+        return context.loadServiceExtensions();
+    }
+
+    /**
+     * Create a {@link ServiceExtensionContext} that will be used in this runtime. If e.g. a custom DI mechanism should be used,
+     * this would likely need to be overridden.
+     *
+     * @param typeManager The TypeManager (for JSON de-/serialization)
+     * @param monitor     a Monitor
+     * @return a {@code ServiceExtensionContext}
+     */
+    @NotNull
+    protected ServiceExtensionContext createContext(TypeManager typeManager, Monitor monitor) {
+        return new DefaultServiceExtensionContext(typeManager, monitor);
+    }
+
+    /**
+     * Hook that is called when a runtime is shutdown (e.g. after a CTRL-C command on a command line). It is highly advisable to
+     * forward this signal to all extensions through their {@link ServiceExtension#shutdown()} callback.
+     *
+     * @param serviceExtensions All extensions that should receive the shutdown signal.
+     * @param monitor           A monitor - should you need one.
+     */
+    protected void shutdown(List<ServiceExtension> serviceExtensions, Monitor monitor) {
+        ListIterator<ServiceExtension> iter = serviceExtensions.listIterator(serviceExtensions.size());
+        while (iter.hasPrevious()) {
+            iter.previous().shutdown();
+        }
+        monitor.info("Connector shutdown complete");
+    }
+
+    /**
+     * Hook point to initialize the vault. By default this calls {@link org.eclipse.dataspaceconnector.system.ExtensionLoader#loadVault(ServiceExtensionContext)}
+     * to instantiate and configure the vault.
+     * <p>
+     * <strong>Do not use this to provide a custom Vault implementation! Use the extension mechanism for that! </strong>
+     *
+     * @param context An {@code ServiceExtensionContext} to resolve the {@code Vault} from.
+     */
+    protected void initializeVault(ServiceExtensionContext context) {
```

Comment:
load or init?

Reply:
again, since the method returns `void`, and since the fact that internally the vault is actually _loaded_ (as opposed to created) is an implementation detail, I think `initialize` is fine here. It implies that the vault is already there beforehand. 
I'll improve the comments however.